update the build related stuff.

yinghuang123 opened the pull request:
small change to metadata.

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below


# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)


-----------------------------------------------------------------------------------------------------------

yinghuang123 committed:
update the build related stuff.
small change to metadata.

yinghuang123 commented:
encountered the playbook1-email id issue.
manually changed that to emailid, since manual involvement, so I also js-beautify the generated mainTemplate.json file.
build validation passed, and manual deployment passed.

according to this document:
    Tools/Create-Azure-Sentinel-Solution/V2/README.md:292 * Open up <https://aka.ms/AzureSentinelPrP> which launches the Azure portal with the needed private preview flags.
it is verified manually again a special uri.

maybe next time, I could try auto deploy from cli to verify.

eerus commented on Solutions/CohesitySecurity/Package/createUiDefinition.json:
```diff
@@ -60,7 +60,17 @@
             "type": "Microsoft.Common.TextBlock",
             "options": {
-              "text": "This product integrates Cohesity Helios with Microsoft Sentinel to stay updated with the security events from your Cohesity environment and immediately respond to a ransomware attack or an anomaly."
+              "text": "This Solution installs the data connector for CohesitySecurity. You can get CohesitySecurity custom log data in your Microsoft Sentinel workspace. Configure and enable this data connector in the Data Connector gallery after this Solution deploys. This data connector creates custom log table(s) Cohesity_CL in your Microsoft Sentinel / Azure Log Analytics workspace."
```
The new text is incorrect. We don't have data connectors and custom tables.
I'd leave the previous text
`This product integrates Cohesity Helios with Microsoft Sentinel to stay updated with the security events from your Cohesity environment and immediately respond to a ransomware attack or an anomaly.`

yinghuang123 replied:
@kishannerella manually change this from the generated json file.
this is incorrect, I need to fix it from the build script.

yinghuang123 replied:
maybe I could also spend some time to fix the email id issue from the build script.

eerus reviewed (changes requested):
LGTM except for 1 text correction.

yinghuang123 committed:
add this file, but which still not working yet.
    ./Solutions/CohesitySecurity/Package/mainTemplate.json.sh
update the build script.
